chemicaltoolbox/openbabel/ob_remIons.py

```diff
@@ -15,20 +16,25 @@ def parse_command_line():
     parser.add_argument('-iformat', default='sdf', help='input file format')
     parser.add_argument('-i', '--input', required=True, help='input file name')
     parser.add_argument('-o', '--output', required=True, help='output file name')
+    parser.add_argument('-idx', default=False, action='store_true', help='should output be an indexed text table? works only for inchi/smiles, otherwise is ignored')
     return parser.parse_args()
 
 
 def remove_ions(args):
-    outfile = pybel.Outputfile(args.iformat, args.output, overwrite=True)
-    for mol in pybel.readfile(args.iformat, args.input):
+    outfile = open(args.output, 'w')
+
+    for index, mol in enumerate(pybel.readfile(args.iformat, args.input)):
         if mol.OBMol.NumHvyAtoms() > 5:
             mol.OBMol.StripSalts(0)
             if 'inchi' in mol.data:
                 del mol.data['inchi']  # remove inchi cache so modified mol is saved
-            # Check if new small fragments have been created and remove them
-            if mol.OBMol.NumHvyAtoms() > 5:
-                outfile.write(mol)
-    outfile.close()
+
+        mol = mol.write(args.iformat) if mol.OBMol.NumHvyAtoms() > 5 else '\n'
+
+        if args.idx and args.iformat in ['inchi', 'smi']:
+            outfile.write(f'{index}\t{mol}')
+        elif mol != '\n':
+            outfile.write(f'{mol}')
```

Comment:
Do you know if this works for other file formats as well? It's a pity this tool doesn't have a test for SD-files.

Reply:
In case of formats that are not one-liners this table setup with columns would not make much sense, so it works only for one-line formats.